src/clib/lib/enkf/obs_vector.cpp

```diff
@@ -460,12 +458,68 @@ obs_vector_type *obs_vector_alloc_from_GENERAL_OBSERVATION(
     }
 }
 
+static bool history_init_ts(const history_source_type history,
+                            const char *summary_key, double_vector_type *value,
+                            bool_vector_type *valid,
+                            const ecl_sum_type *refcase) {
+    bool initOK = false;
+
+    double_vector_reset(value);
+    bool_vector_reset(valid);
+    bool_vector_set_default(valid, false);
```

Comment:
Should probably use `std::optional` to avoid manual memory management:
```cpp
static bool history_init_ts(const history_source_type history,
                            const char *summary_key, double_vector_type *value,
                            bool_vector_type *valid,
                            const ecl_sum_type *refcase) {
    bool initOK = false;

    double_vector_reset(value);
    bool_vector_reset(valid);
    bool_vector_set_default(valid, false);

    std::optional<std::string> local_key = std::nullopt;
    if (history == REFCASE_HISTORY) {
        /* Must create a new key with 'H' for historical values. */
        const ecl_smspec_type *smspec = ecl_sum_get_smspec(refcase);
        const char *join_string = ecl_smspec_get_join_string(smspec);
        ecl_smspec_var_type var_type =
            ecl_smspec_identify_var_type(summary_key);

        if ((var_type == ECL_SMSPEC_WELL_VAR) ||
            (var_type == ECL_SMSPEC_GROUP_VAR))
            local_key = std::string(std::move(util_alloc_sprintf(
                "%sH%s%s", ecl_sum_get_keyword(refcase, summary_key),
                join_string, ecl_sum_get_wgname(refcase, summary_key))));
        else if (var_type == ECL_SMSPEC_FIELD_VAR)
            local_key = std::string(std::move(util_alloc_sprintf(
                "%sH", ecl_sum_get_keyword(refcase, summary_key))));
    } else
        local_key = std::string(summary_key);

    if (local_key.has_value()) {
        if (ecl_sum_has_general_var(refcase, local_key.value().c_str())) {
            for (int tstep = 0; tstep <= ecl_sum_get_last_report_step(refcase);
                 tstep++) {
                if (ecl_sum_has_report_step(refcase, tstep)) {
                    int time_index = ecl_sum_iget_report_end(refcase, tstep);
                    double_vector_iset(
                        value, tstep,
                        ecl_sum_get_general_var(refcase, time_index,
                                                local_key.value().c_str()));
                    bool_vector_iset(valid, tstep, true);
                } else
                    bool_vector_iset(valid, tstep,
                                     false); /* Did not have this report step */
            }
            initOK = true;
        }
    }
    return initOK;
}
```

Reply:
Looks good!